scripts/manageacl.py

```diff
@@ -0,0 +1,90 @@
+from azure.search.documents import SearchClient
+from azure.search.documents.indexes import SearchIndexClient
+from azure.search.documents.indexes.models import (
+    SimpleField,
+    SearchFieldDataType
+)
+import argparse
+from azure.identity import AzureDeveloperCliCredential
+from azure.core.credentials import AzureKeyCredential, TokenCredential
+
+def enable_acls(credential: AzureKeyCredential|TokenCredential, search_service: str, index: str):
+    index_client = SearchIndexClient(
+        endpoint=f"https://{args.search_service}.search.windows.net/",
+        credential=credential)
+    if args.verbose: print(f"Enabling acls for index {index}")
+    try:
+        index_definition = index_client.get_index(index)
+        if not any(field.name == 'oids' for field in index_definition.fields):
+            index_definition.fields.append(SimpleField(name="oids", type=SearchFieldDataType.Collection(SearchFieldDataType.String), filterable=True))
+        if not any(field.name == 'groups' for field in index_definition.fields):
+            index_definition.fields.append(SimpleField(name="groups", type=SearchFieldDataType.Collection(SearchFieldDataType.String), filterable=True))
+        
+        index_client.create_or_update_index(index_definition)
+    except Exception as e:
+        print(f"\tGot an error while updating index {index} -> {e} --> unable to enable acls")
+
+def update_acl(credential: AzureKeyCredential|TokenCredential, search_service: str, index: str, acl_type: str, acl_action: str, document: str, acl: str = None):
+    search_client = SearchClient(
+        endpoint=f"https://{args.search_service}.search.windows.net/",
```

Comment:
Please avoid using args in the functions as it makes code much harder to test (prepdocs does this a bunch now, and its made it hard to write tests). Please pass as args or create a class which takes in the configuration if there are too many for args.

Reply:
Do you have an example of how this should be done? i'm a bit confused as how i should write this, there's an argparse at the beginning of the script and then smaller functions.

Reply:
making these adjustments